src/Auth0Client.ts

```diff
@@ -984,12 +958,16 @@ export class Auth0Client {
     options: GetTokenWithPopupOptions = {},
     config: PopupConfigOptions = {}
   ) {
-    options.audience = options.audience || this.options.audience;
+    if (!options.authorizationParams) {
+      options.authorizationParams = {
+        audience: this.options.authorizationParams.audience
+      };
+    }
```

Comment:
Is there a reason we explicitly check if `authorizationParams` is undefined, while we dont in for example `getTokenSilently` ?

Would adding something like below to the top of the functions that access authorizationParams work ?
```
options.authorizationParams = options.authorizationParams || {}
```

Reply:
This was added because it's the only function where `authorizationParams` is actually required to be set as it assigns to it. The other functions either construct the object themselves (like `getTokenSilently`) or just read from it.

Maybe this could change to be more like `getTokenSilently` so that it reads a little better?

Reply:
What would you say about seeing if we can have the same approach in all these methods for the case where authorizationParams or logoutParams was not specified, makes it easier to understand that's just ensuring the objects exists. 

Could be what I mentioned above, or anything that does the job that makes it clearer.

Reply:
Improved the handling around this